tests/course/test_review_view.py

```diff
@@ -6,47 +6,63 @@
 
 @pytest.mark.django_db
 class TestReviewView:
-    def test_create_review(self, client, persist_user, persist_review):
-        client.force_login(persist_user)
-        response = client.post(reverse('create_review', args=[persist_review.course_id]), {
-            'rating': persist_review.rating,
-            'content': persist_review.content
-        })
+    def test_add_review(self, authorized_second_client, new_review):
+        course_response = authorized_second_client.get('/course/' + str(new_review.course_id))
+        assert course_response.status_code == 200
+        response = authorized_second_client.post(f'/course/{new_review.course_id}/add_review',
+                                                 {'rating': new_review.rating, 'content': new_review.content})
         assert response.status_code == 302
         assert Review.objects.exists()
 
-    def test_invalid_review_message(self, client, persist_user, persist_review):
-        client.force_login(persist_user)
-        response = client.post(reverse('create_review', args=[persist_review.course_id]), {
-            'content': persist_review.content
-        })
+    def test_invalid_review_message(self, authorized_second_client, new_review):
+        response = authorized_second_client.post(f'/course/{new_review.course_id}/add_review',
+                                                 {'content': new_review.content})
         messages = list(get_messages(response.wsgi_request))
         assert len(messages) == 1
         assert str(messages[0]) == "You must enter rating"
 
-    def test_update_review(self, client, persist_user, persist_review):
-        client.force_login(persist_user)
-        response = client.post(reverse('update_review', args=[persist_review.pk]),
-                               data={'rating': persist_review.rating, 'content': persist_review.content})
+    def test_display_review_existed(self, authorized_second_client, persist_review):
+        response = authorized_second_client.get('/course/' + str(persist_review.course_id), {'review_view': True})
+        assert response.status_code == 200
+        assert Review.objects.count() == 1
+        assert b"Update Review" in response.content
+        assert b"Delete Review" in response.content
+
+    def test_display_review_not_existed(self, authorized_second_client, new_review):
+        response = authorized_second_client.get('/course/' + str(new_review.course_id), {'review_view': False})
+        assert response.status_code == 200
+        assert Review.objects.count() == 0
+        assert b"Add Review" in response.content
```

Comment:
I'd also assert the other options are not present

Reply:
Same as the comment above

Reply:
Done